core/src/main/java/cz/o2/proxima/repository/ConfigRepository.java

```
@@ -293,6 +298,23 @@ private ConfigRepository(
     } catch (Exception ex) {
       throw new IllegalArgumentException("Cannot read config settings", ex);
     }
+
+    if (cachingEnabled) {
+      if (cachedConfigConstructed != null) {
```

is this necessary?

---

I mean, shouldn't we always fail fast?